src/main/java/org/folio/dcb/domain/mapper/TransactionMapper.java

```diff
@@ -0,0 +1,32 @@
+package org.folio.dcb.domain.mapper;
+
+import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.domain.dto.DcbTransaction;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TransactionMapper {
```

Comment:
I believe, here could be applied auto mapper generation instead of manual implementation.
see here for the details
https://github.com/folio-org/mod-dcb/pull/10/files#diff-680f8c028656660c29f7edcb3a2cab698f68d912b5ecb3bdaec924fb0ed937e6

Reply:
drilling down into the code...
the way, you implemented mapper is the proper one.
I missed it, you're producing entity from the complex dto, and no direct mapping is possible.

sorry for confusing you.